LanguageFeatures/Augmentations/augmenting_functions_A04_t29.dart

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that an augmenting function or constructor's signature
+/// matches if:
+/// - It has the same number of type parameters with the same type parameter
+///   names (same identifiers) and bounds (after type annotation inheritance),
+///   if any (same types, even if they may not be written exactly the same in
+///   case one of the declarations needs to refer to a type using an import
+///   prefix).
+/// - The return type (if not omitted) is the same as the augmented
+///   declaration's return type.
+/// - It has the same number of positional and optional parameters as the
+///   augmented declaration.
+/// - It has the same set of named parameter names as the augmented declaration.
+/// - For all corresponding pairs of parameters:
+///   - They have the same type (if not omitted in the augmenting declaration).
+///   - They have the same `required` and `covariant` modifiers.
+/// - For all positional parameters:
+///   - The augmenting function's parameter name is `_`, or
+///   - The augmenting function's parameter name is the same as the name of the
+///     corresponding positional parameter in every preceding declaration that
+///     doesn't have `_` as its name.
+/// ...
+/// It's a compile-time error if:
+/// - The signature of the augmenting function does not match the signature of
+///   the augmented function.
+///
+/// @description Checks that if the name of a positional parameter was augmented
+/// to `_` then it is a compile-time error to use an old name.
```

Comment:
The wildcard doesn't do anything to the name of the parameter, it indicates that the author of this declaration in the augmentation chain doesn't want to access this parameter (so we're _omitting_ the name rather than changing or augmenting it). This also means that `x` or `_x` below isn't an old name, it is _the_ name of the parameter. But it is correct that it is a compile-time error to use the name of the parameter in a declaration in the augmentation chain for that function/method whose declaration of that parameter is a wildcard.